lib/rest/routes/credentials.js

```diff
@@ -83,10 +78,16 @@ module.exports = function () {
   });
 
   router.get('/:consumerId', function (req, res, next) {
-    credentialSrv
-      .getCredentials(req.params.consumerId)
-      .then(credentials => res.json({ credentials }))
-      .catch(next);
+    findConsumer(req.params.consumerId)
+      .then((consumer) => {
+        if (!consumer) {
+          return res.status(422).json(new Error('Consumer Not Found: id:' + req.body.consumerId));
```

Comment:
422 is kind of form validation. it should be 404, since no consumer exists

Reply:
That's absolutely correct. 422 means "not processable" entity.